.github/workflows/test.yml

```diff
@@ -0,0 +1,55 @@
+name: Test
+
+on:
+    pull_request:
+        types:
+            - opened
+            - synchronize
+            - reopened
+    push:
+        branches:
+            - main
+            - next
+
+jobs:
+    test:
+        name: Test
+        runs-on: ubuntu-latest
+        steps:
+            - run: echo "${{ github.actor }}"
+
+            - uses: actions/checkout@v3
+              with:
+                  token: ${{ secrets.GITHUB_TOKEN }}
+            - run: |
+                  git config user.name github-actions
+                  git config user.email github-actions@github.com
+
+            - uses: pnpm/action-setup@v2
+              with:
+                  version: 8
+
+            - name: Use Node.js 18.x
+              uses: actions/setup-node@v3
+              with:
+                  node-version: 18
+                  registry-url: "https://registry.npmjs.org"
+                  cache: "pnpm" # https://github.com/actions/setup-node/blob/main/docs/advanced-usage.md#caching-packages-dependencies
+
+            - run: pnpm install --frozen-lockfile
+
+            - name: Copy schema files
+              run: pnpm run copy-schema-files
+
+            - name: Build
+              run: pnpm run build:packages
```

Comment:
It's odd that we have to build the packages before running the tests, but I don't think there's another way (expect exhaustive mocking, which I'm not sure we actually want)